### ISSUE TITLE
Can we perform fusedBatchNorm using Direct Convolution Implementations? 

### ISSUE THREAD

**Issue by Alok-Ranjan23** [oneapi-src/oneDNN#757]
Hi ,
   Can we perform convolutionWithFusedBatchNorm using direct convolution implementation like relu operation is fused? Now, i am using convolution followed by  a postOps(which performs batch_normalization) in case of direct convolution implementations that is an overhead. Is there any way to perform fusedBatchNorm with convolution using direct Convolution implementation path?

**Comment by emfomenk** [comment]
Hi @Alok-Ranjan23,

Do you do training (in which case BNorm needs to compute mean and variance of the input tensor) or inference (in which case BNorm is just scale and shift)?
- In the former case there is no way to fuse convolution and batch norm.
- In the latter case currently the only way of doing the fusion is to embed the batch normalization into convolution's weights and bias:
    ```
    Bnorm(Conv(X, W, bias)) := scale * Conv(X, W, bias) + shift =
    = Conv(X, scale * W, scale * bias) + shift = 
    = Conv(X, scale * W, scale * bias + shift)
    ```

    We also plan to add binary post operations, so that scale and shift fusion would be more user friendly. However, this feature is few released ahead.

**Comment by emfomenk** [comment]
Closing due to lack of activity.
Feel free to reopen or open a new one.
